scripts/wrappers/reset.py

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/python3
+import click
+import os
+import subprocess
+import sys
+import time
+
+from common.utils import (
+    get_available_addons,
+    get_current_arch,
+    get_status,
+    snap_data,
+    wait_for_ready,
+    snap_common,
+    is_cluster_locked,
+    exit_if_no_permission,
+    ensure_started,
+)
+
+
+KUBECTL = os.path.expandvars("$SNAP/microk8s.kubectl-wrapper")
+
+
+def exit_if_multinode():
+    """
+    Exit if we cannot get the list of nodes or if we are in a multinode cluster
+    """
+    cmd = [KUBECTL, "get", "no", "-o", "name"]
+    res = run_silently(cmd)
+    if not res:
+        print("Failed to query the cluster nodes.")
+        sys.exit(1)
+    nodes = res.split()
+    if len(nodes) > 1:
+        print(
+            "This is a multi-node MicroK8s deployment. Reset is applicable for single node clusters."
+        )
+        print("Please remove all joined nodes before calling reset.")
+        sys.exit(0)
+
+
+def disable_addon(repo, addon, args=[]):
+    """
+    Try to disable an addon. Ignore any errors and/or silence any output.
+    """
+    wait_for_ready(timeout=30)
+    script = snap_common() / "addons" / repo / "addons" / addon / "disable"
+    if not script.exists():
+        return
+
+    subprocess.run([script, *args], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
```

Comment:
One final suggestion for this, since it can still raise exceptions like `PermissionsDenied` if executable bits are missing:

```suggestion
    try:
        subprocess.run([script, *args], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
    except (FileNotFoundError, PermissionDenied, subprocess.CalledProcessError) as e:
        print(f"Ignoring error: {e}", file=sys.stderr)
```

With the `FileNotFoundError`, the `script.exists()` check above could optionally be removed as well.